src/transformers/trainer.py

```diff
@@ -811,6 +816,11 @@ def train(
             num_train_epochs = 1
             num_update_steps_per_epoch = max_steps
 
+        if self.args.warmup_ratio > 0 and self.args.warmup_steps == 0:
+            if self.args.warmup_ratio > 1:
+                raise ValueError("warmup_ratio must lie in range [0,1]")
```

Comment:
This part should go in an `elif` at init: it's best to detect invalid arguments at the very beginning.